tests/test_discovery.py

```diff
@@ -413,6 +416,103 @@ def test_userip_missing_is_not_added_to_discovery_uri(self):
       self.assertEqual(e.uri, 'http://example.com')
 
 
+class DiscoveryFromAppEngineCache(unittest.TestCase):
+  def test_appengine_memcache(self):
+    # Hack module import
+    self.orig_import = __import__
+    self.mocked_api = mock.MagicMock()
+
+    def import_mock(name, *args):
+      if name == 'google.appengine.api':
+        return self.mocked_api
+      return self.orig_import(name, *args)
+
+    import_fullname = '__builtin__.__import__'
+    if sys.version_info[0] >= 3:
+      import_fullname = 'builtins.__import__'
+
+    with mock.patch(import_fullname, side_effect=import_mock):
+      ns = 'google-api-client'
```

Comment:
Please avoid initialisms and acronyms in identifiers for the same reasons single-letter names should be avoided.